### PR TITLE
Foolmoon

### PR DESCRIPTION
**Pull request by tituswhang**

### CODE REVIEW

File: Assets/Darwin/DarwinAI.cs [hunk 1]
```diff
@@ -66,33 +98,50 @@ float getAngle(GameObject other)
...
+
+        // The distance from the closest object. Set as 10.0f which is the radius of the triggerArea.
         float dist = 10.0f;
 
-        if(darwins.Count == 1) // if frog is still hungry, search for more food
+        // This if-statement has darwins.Count as 1 as it also counts the current Darwin which is technically within
+        // the triggerArea. If Darwin is still hungry, search for more food
+        if (darwins.Count == 1)
         {
-            dist = float.MaxValue;
```
Comment (mis825) [inline Assets/Darwin/DarwinAI.cs:76]: Why is line 76 being removed?
`dist = float.MaxValue;`

File: Assets/Darwin/DarwinAI.cs [hunk 2]
```diff
@@ -8,55 +10,85 @@
...
+
+        // TriggerArea of the current Darwin. It belongs to the Darwin as a chlid
         triggerArea = gameObject.GetComponentInChildren<TriggerArea>();
 
         rigidBody.mass = traits.Mass;
     }
 
     void Update()
     {
-        if(traits.Energy <= 0)
```
Comment (mis825) [inline Assets/Darwin/DarwinAI.cs:27]: Is `traits.Energy <= 0` the same as `!traits.isLiving()`?

**Review by mis825 — CHANGES_REQUESTED** [review]
Would it be feasible to remove the purple lines that remain still for more than 10 seconds? Sometimes the purple lines permanently show on the screen without the Darwin.
image

File: Assets/Darwin/DarwinAI.cs (hunk 2 above)
Comment (tituswhang) [reply Assets/Darwin/DarwinAI.cs:27]: Yes. This change is being made to use the `isLiving()` from DarwinTraits

File: Assets/Darwin/DarwinAI.cs (hunk 1 above)
Comment (tituswhang) [reply Assets/Darwin/DarwinAI.cs:76]: I did `dist = float.MaxValue` to use `dist` as a comparison to all the distances between the current Darwin and every other object around it. If an object is closer than the previous object, set `dist` as the distance between the next object. However, I have found a better, efficient solution to keep track of all the objects that were around the current Darwin.

**Comment by tituswhang** [comment]
> Would it be feasible to remove the purple lines that remain still for more than 10 seconds? Sometimes the purple lines permanently show on the screen without the Darwin. image

I haven't gotten to replicating that issue on my machine. I will try to fix it if I encounter it.

**Review by mis825 — APPROVED** [review]

**Pull request merged**